jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeObjectNodeTest.kt

```diff
@@ -0,0 +1,284 @@
+                    .versions(createS3ObjectVersion("my/folder/picture.png", "null"))
+                    .build()
+        }
+        val bucket = S3VirtualBucket(s3Bucket, s3Client)
+        val parent = S3TreeDirectoryNode(bucket, null, "my/folder/")
+        val sut = S3TreeObjectNode(parent, "my/folder/picture.png", objectSize, lastModifiedTime)
+        sut.showHistory = true
+
+        assertThat(sut.children).isEmpty()
+    }
+
+    @Test
+    fun `get children with more loaded`() {
+        val requestCaptor = argumentCaptor<ListObjectVersionsRequest>()
+        val s3Client = delegateMock<S3Client> {
+            on { listObjectVersions(requestCaptor.capture()) }.thenReturn(
+                ListObjectVersionsResponse.builder()
+                    .versions(createS3ObjectVersion("my/folder/picture.png", "1"), createS3ObjectVersion("my/folder/picture.png", "2"))
+                    .nextKeyMarker("KeyToken")
+                    .nextVersionIdMarker("VersionToken")
+                    .isTruncated(true)
+                    .build(),
+                ListObjectVersionsResponse.builder()
+                    .versions(createS3ObjectVersion("my/folder/picture.png", "3"))
+                    .build()
+            )
+        }
+
+        val bucket = S3VirtualBucket(s3Bucket, s3Client)
+        val parent = S3TreeDirectoryNode(bucket, null, "my/folder/")
+        val sut = S3TreeObjectNode(parent, "my/folder/picture.png", objectSize, lastModifiedTime)
+        sut.showHistory = true
+
+        assertThat(sut.children).containsExactly(
+            S3TreeObjectVersionNode(sut, "1", objectSize, lastModifiedTime),
+            S3TreeObjectVersionNode(sut, "2", objectSize, lastModifiedTime),
+            S3TreeContinuationNode(bucket, sut, sut.key, VersionContinuationToken("KeyToken", "VersionToken"))
+        )
+
+        (sut.children.last() as S3TreeContinuationNode<*>).loadMore()
+
+        assertThat(sut.children).containsExactly(
+            S3TreeObjectVersionNode(sut, "1", objectSize, lastModifiedTime),
+            S3TreeObjectVersionNode(sut, "2", objectSize, lastModifiedTime),
+            S3TreeObjectVersionNode(sut, "3", objectSize, lastModifiedTime)
+        )
+
+        assertThat(requestCaptor.allValues).hasSize(2)
+        val firstRequest = requestCaptor.firstValue
+        assertThat(firstRequest.bucket()).isEqualTo("foo")
+        assertThat(firstRequest.prefix()).isEqualTo("my/folder/picture.png")
+        assertThat(firstRequest.keyMarker()).isNull()
+        assertThat(firstRequest.versionIdMarker()).isNull()
+
+        val secondRequest = requestCaptor.secondValue
+        assertThat(secondRequest.bucket()).isEqualTo("foo")
+        assertThat(secondRequest.prefix()).isEqualTo("my/folder/picture.png")
+        assertThat(secondRequest.keyMarker()).isEqualTo("KeyToken")
+        assertThat(secondRequest.versionIdMarker()).isEqualTo("VersionToken")
+    }
+
+    @Test
+    fun `load more is idempotent`() {
```

Comment:
worth asserting that we'll never spin off multiple requests in response to multiple "load mores"?

Reply:
Good call, will get that added to both nodes